dev/context.go

```diff
@@ -0,0 +1,26 @@
+package dev
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+)
+
+type controllerContextKey int
+
+const loggerContextKey controllerContextKey = iota
+
+// Add a logger to the given context, to make it easier passing it around.
+func ContextWithLogger(parent context.Context, logger logr.Logger) context.Context {
+	return context.WithValue(context.Background(), loggerContextKey, logger)
+}
+
+// Get the logger from the given context.
+// Returning a discarding logger, if none was set.
+func LoggerFromContext(ctx context.Context) logr.Logger {
+	logger, ok := ctx.Value(loggerContextKey).(logr.Logger)
+	if ok {
+		return logger
+	}
+	return logr.Discard()
```

Comment:
If we don't return some implementation of the interface, the caller would have to do an extra nil/ok-check.
Because we ideally want to log in a lot of places, we have to check everywhere.

Reply:
the discard will discard the messages here https://github.com/go-logr/logr/blob/master/discard.go#L19

Reply:
as a default option should we do something like this https://github.com/go-logr/logr/blob/master/example_test.go#L28

Reply:
you are right it make sense as it is